File: languages/js/src/entity/integrity/network.ts
import * as proto from "../../bridge/proto/bloock_config";

/**
 * Represents a network.
 */
export enum Network {
  UNRECOGNIZED = -1,
  ETHEREUM_MAINNET = 0,
  ETHEREUM_SEPOLIA = 1,
  GNOSIS_CHAIN = 2,
  POLYGON_CHAIN = 4
}

export namespace Network {
  export function toProto(network: Network): proto.Network {
    switch (network) {
      case Network.ETHEREUM_MAINNET:
        return proto.Network.ETHEREUM_MAINNET;
      case Network.ETHEREUM_SEPOLIA:
        return proto.Network.ETHEREUM_SEPOLIA;
      case Network.GNOSIS_CHAIN:
        return proto.Network.GNOSIS_CHAIN;
      case Network.POLYGON_CHAIN:
        return proto.Network.POLYGON_CHAIN;
      default:
        return proto.Network.UNRECOGNIZED;
    }
  }

  export function fromProto(network: proto.Network | undefined): Network {
    switch (network) {
      case proto.Network.ETHEREUM_MAINNET:
        return Network.ETHEREUM_MAINNET;
      case proto.Network.ETHEREUM_SEPOLIA:
        return Network.ETHEREUM_SEPOLIA;
      case proto.Network.GNOSIS_CHAIN:
        return Network.GNOSIS_CHAIN;
      case proto.Network.POLYGON_CHAIN:
        return Network.POLYGON_CHAIN;
      default:
        return Network.UNRECOGNIZED;
    }
  }
}

File: languages/go/entity/integrity/network.go
package integrity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// Network represents a network.
type Network = proto.Network

func NetworkToProto(network Network) *proto.Network {
	if network == -1 {
		return nil
	}
	return &network
}

// networks represents a set of predefined networks.
type networks struct {
	EthereumGnosis  Network
	EthereumMainnet Network
	EthereumSepolia Network
	EthereumPolygon Network
}

// ListOfNetworks returns a networks instance with predefined network values.
func ListOfNetworks() networks {
	return networks{
		EthereumGnosis:  proto.Network_GNOSIS_CHAIN,
		EthereumMainnet: proto.Network_ETHEREUM_MAINNET,
		EthereumSepolia: proto.Network_ETHEREUM_SEPOLIA,
		EthereumPolygon: proto.Network_POLYGON_CHAIN,
	}
}

// AnchorParams represents parameters for anchor-related operations.
type AnchorParams struct {
	Timeout int64
}

// NewAnchorParams creates a new AnchorParams instance with default values.
func NewAnchorParams() AnchorParams {
	return AnchorParams{}
}

// NetworkParams represents parameters for network-related operations.
type NetworkParams struct {
	Network Network
}

// NewNetworkParams creates a new NetworkParams instance with default values.
func NewNetworkParams() NetworkParams {
	return NetworkParams{
		Network: -1,
	}
}

const (
	InvalidRecordError     = "Record error: Invalid record"
	RecordNotFoundError    = "Infrastructure error: Http Client error: Request error - API connected by HttpClient found an error: record not found"
	WaitAnchorTimeoutError = "Anchor error: Wait Anchor timed out"
	AnchorNotFoundError    = "Infrastructure error: Http Client error: API connected by HttpClient found an error: anchor not found"
)
